### DISCUSSION TITLE
Is there a guide for end users?

### DISCUSSION THREAD

**Discussion by yuchen-sun** [cloud-hypervisor/cloud-hypervisor#3373] [Q&A]
We are considering using cloud-hypervisor to build our next generation application. But we don't find out how to use cloud-hypervisor, should we use api directly or use libvirtd or something like that to manage it?
Any questions or advises, please don't hesitate to share. Thank you!

**Comment by bryteise** [comment]
The libvirt support for cloud-hypervisor is still early and doesn't handle all the use cases the cloud-hypervisor API supports.

**Comment by yuchen-sun** [comment]
Thanks! @bryteise 
We will use cloud-hypervisor api instead of libvirtd
Also, does a cloud-hypervisor only handle one vm?

**Comment by bryteise** [comment]
Yes, each cloud-hypervisor instance runs a single VM.

**Comment by yuchen-sun** [comment]
And is there any management experience when a host has multiple VMs?

**Reply by sboeuf** [reply]
Well either you can have your own software doing the management of all your VMs, or you can use libvirt.
In case we're talking about running your application in containers, and if you want to leverage the virtualization layer provided by Cloud Hypervisor, one alternative would be to use Kata Containers in the context of Kubernetes, so that VMs are created by Cloud Hypervisor under the hood.
